### PR TITLE
PKCE OAuth support added

### PR DESCRIPTION
**Pull request by hareskoug**
Do some general code cleanup and add support for PKCE for three-legged OAuth.

### CODE REVIEW

File: visier/connector/sessions.py [hunk 1]
```diff
@@ -98,7 +100,7 @@ class VisierSession:
     def __init__(self, auth: Authentication) -> None:
         self._auth = auth
         self._session = None
-        self._timeout = 30
+        self._timeout = 300
```
Comment (cszhu) [inline visier/connector/sessions.py:103]: Is there a reason why this was increased by 10? Or was it a typo before?

File: visier/connector/sessions.py [hunk 2]
```diff
@@ -218,31 +220,43 @@ def _connect_auth_code(self, auth: OAuth2):
         """Connect to Visier using (three-legged) OAuth2.
         This method will attempt to open a browser for the authentication and consent screens.
         It will also spin up a local web server to receive the OAuth2 authorization code."""
-        url_prefix = auth.host + "/v1/auth/oauth2"
+        def mk_pkce_pair():
```
Comment (cszhu) [inline visier/connector/sessions.py:223]: For the life of me I can't exactly figure out what `mk_pkce_pair` stands for - maybe folks who are using this code would know better than me, but maybe can spell out some of it for readability reasons, besides that this is fine.

File: visier/connector/sessions.py [hunk 3]
```diff
@@ -218,31 +220,43 @@ def _connect_auth_code(self, auth: OAuth2):
...
-                redirect_uri_arg = "" # Empty means use the redirect_uri registered with Visier
-            browser_url = f'{url_prefix}/authorize?apikey={auth.api_key}&response_type=code&client_id={auth.client_id}{redirect_uri_arg}'
+                query_args["redirect_uri"] = auth.redirect_uri
+
+            browser_url = f'{url_prefix}/authorize?{urlencode(query_args)}'
             # Launch the browser for authentication and consent screens
             webbrowser.open(browser_url)
             try:
                 # Wait up to 2 minutes for the user to complete the OAuth2 code flow
                 code = svr.queue.get(block=True, timeout=120)
```
Comment (cszhu) [inline visier/connector/sessions.py:258]: Just curious why this timeout is different than the `self. _timeout` specified earlier; if it's just arbitrary then that's fine too.

**Review by cszhu — APPROVED** [review]
Just a few questions, but overall lgtm.

File: visier/connector/sessions.py (hunk 1 above)
Comment (hareskoug) [reply visier/connector/sessions.py:103]: I had lowered it before to test OAuth callback expiry and forgot to change it back to 5min. 😞

File: visier/connector/sessions.py (hunk 2 above)
Comment (hareskoug) [reply visier/connector/sessions.py:223]: Ok. Will add docstring

File: visier/connector/sessions.py (hunk 1 above)
Comment (hareskoug) [reply visier/connector/sessions.py:103]: No, that should be lowered back. It was increased to support backend debugging

File: visier/connector/sessions.py (hunk 3 above)
Comment (hareskoug) [reply visier/connector/sessions.py]: The `_timeout` is a per request timeout. The 2min timeout is for the end-to-end authorization code flow. I've refactored this slightly to make that part clearer.

**Pull request merged**